File: AEDSII-TP/tp1/tp01/TP01Q11 - RECURSIVO - Palíndromo em Java/RECURSIVOpalindromo.java
/*
 * Nome : Diego Da Silva Ferreira
 * Matricula : 71 06 78
 *
 * */


public class RECURSIVOpalindromo {

	/*
	 * verifica se um string é um palindromo 
	 * @param String s - string a ser analizada
	 * @param int cabecote - percorre a string (utilizado na rcursividade)
	 * @return boolean resp - true cao a string seja um palindromo
	 * */
	public static boolean isPalindromo (String s , int cabecote){
	
		boolean resp = true;

		if(cabecote < (int) (s.length() /2) ){
			if(s.charAt(cabecote) == s.charAt((s.length() - 1) - cabecote))
				resp = isPalindromo(s , cabecote + 1);
			else 
				resp = false;
		}//end if
			
		return resp;
	}//end isPalindromo


	/*
	 * chama a função recursiva isPalindromo 
	 * @param String s - string a ser analizada 
	 * @return boolean - true caso a string seja um palindromo
	 * */
	public static boolean isPalindromo (String s){
	
		return isPalindromo(s , 0);
	}//end isPalindromo


	public static void main (String[] args) {
	
		String s = MyIO.readLine();

		//metodo para verificar se string é igual a "FIM"
		while(!(s.length() == 3 && 
			s.charAt(0) == 'F' && 
			s.charAt(1) == 'I' && 
			s.charAt(2) == 'M')){

			if(isPalindromo(s)){
				System.out.println("SIM");
			}
			else{
				System.out.println("NAO");
			}	

			s = MyIO.readLine();
		
		}//end while	
	}//end 
}//end RECURSIVOpalindromo

File: AEDSII-TP/tp1/tp01/TP01Q12 - RECURSIVO - Palíndromo em C/RECURSIVOpalindromo.cpp
/*
 * Nome : Diego Da Silva Ferreira
 * Matricula : 71 06 78
 * 
 * */

#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>


/*
 * verifica se uma cadeia de caracteres é um palindromo
 * @param char *s - cadeia de caracteres a ser verificada
 * @param int cabecote - percorre a cadeia a de caracteres (utilizada na recursividade)
 * @return bool resp - true caso a cadeia de caracteres seja um palindromo
 * */
bool isPalindromo (char *s, int cabecote) {

	bool resp = true;

	if(cabecote < (int)(strlen(s) / 2)){
	
		if(s[cabecote] == s[ (strlen(s) - 1) - cabecote] ){
		
			resp = isPalindromo(s , cabecote + 1);
		}
		else{
		
			resp = false;
		}
	}//end if

	return resp;

}//end isPalindromo


int main(void){

	char *s = (char *) calloc (1001 , sizeof (char));

	s[1000] = '\0';
	fgets (s , 999 , stdin);
	s[strlen(s) - 1] = '\0';

	while(strcmp( s , "FIM") != 0){

		if(isPalindromo(s , 0)){
			printf("SIM\n");
		}
		else{
			printf("NAO\n");
		}

		fgets(s , 999, stdin);
		s[strlen(s) - 1] = '\0';

	}//end while

	return 0;
}//end main
